app/controllers/user.rb

```diff
@@ -0,0 +1,36 @@
+get '/' do
+  erb :index
+end
+
+
+post '/login' do #find user, start session
+  @user = User.find_by(username: params[:username], password: params[:password])
+  session[:user_id] = @user.id
```

Comment:
If line 7 doesn't find a user, @user = nil and #id will be an invalid method for nil.